File: public/dsa/arrays/pascals_triangle/recursive/pascals_triangle.py


from typing import List


class Solution:
    def pascalTriangleRow(self, rowNo: int) -> List[int]:
        current_row = []
        prev_row = []
        current_row.append(1)
        if rowNo == 1:
            return current_row
        prev_row = self.pascalTriangleRow(rowNo - 1)
        for i in range(1, len(prev_row)):
            current_row.append(prev_row[i-1] + prev_row[i])
        current_row.append(1)
        return current_row


# tests-start
rowNo = 5
sol = Solution()
print(sol.pascalTriangleRow(rowNo))
# tests-end

'''output
[1, 4, 6, 4, 1]
'''

File: public/dsa/arrays/pascals_triangle/recursive/pascals_triangle.cpp

#include <bits/stdc++.h>
using namespace std;

vector<int> pascalTriangleRow(int rowNo) {
    vector<int> currentRow, prevRow;
    currentRow.push_back(1);
    if (rowNo == 1) {
        return currentRow;
    }
    prevRow = pascalTriangleRow(rowNo - 1);
    for (int i = 1; i < prevRow.size(); i++) {
        currentRow.push_back(prevRow[i - 1] + prevRow[i]);
    }
    currentRow.push_back(1);
    return currentRow;
}

// tests-start
int main() {
    int rowNo = 5;
    auto row = pascalTriangleRow(rowNo);
    for (int v : row) {
        cout << v << " ";
    }
    cout << endl;
    return 0;
}
// tests-end

/*output
1 4 6 4 1
*/